android/src/main/java/com/reactnativemembrane/MembraneModule.kt

```diff
@@ -231,42 +231,38 @@ class MembraneModule(reactContext: ReactApplicationContext) :
     promise.resolve(null)
   }
 
-  @ReactMethod
-  fun updateVideoTrackMetadata(metadata: ReadableMap, promise: Promise) {
-    val trackId = localVideoTrack?.rtcTrack()?.id() ?: return
-    room?.updateTrackMetadata(trackId, metadata.toMap())
+  private fun updateTrackMetadata(trackId: String, metadata: Metadata) {
+    room?.updateTrackMetadata(trackId, metadata)
     val id = localParticipantId ?: return
     val participant = participants[id] ?: return
-    participants[id] = participant.copy(videoTrackMetadata = metadata.toMap())
+    participant.tracksMetadata[trackId] = metadata
     emitParticipants()
+  }
+
+  @ReactMethod
+  fun updateVideoTrackMetadata(metadata: ReadableMap, promise: Promise) {
+    val trackId = localVideoTrack?.rtcTrack()?.id() ?: return
+    updateTrackMetadata(trackId, metadata.toMap())
     promise.resolve(null)
   }
 
   @ReactMethod
   fun updateAudioTrackMetadata(metadata: ReadableMap, promise: Promise) {
```

Comment:
I think this way it might be more difficult for the users of the SDK. The user would need to find the track id (somewhere in the tracks returned by useParticipants) and pass it here. For android/ios sdks getting the track id is easier because the user creates the local tracks. 
By using `updateVideoTrackMetadata` user doesn't need to find the track id, doesn't need to think. They want to update video track metadata, they call `updateVideoTrackMetadata` function :D 
It's more writing for us but the library is supposed to be used more times than it's written.